src/game.cpp

```diff
@@ -7968,8 +7985,85 @@ bool pet_menu(monster *z)
 
         return true;
     }
+    if ( inject == choice ) {
+        auto filter = [] (const item &it) {
+            return it.has_flag("MUTAGEN_CEPHALOPOD") ||
+                   it.has_flag("PURIFIER");
+        };
+        int pos = g->inv_for_filter( _("Injectable mutagen:"), filter );
+        if ( pos == INT_MIN ) {
+            add_msg( _("Never mind.") );
+            return true;
+        }
 
-    return true;
+        item *it = &g->u.i_at(pos);
+
+        if ( !( it->has_flag("MUTAGEN_CEPHALOPOD") ) &&
+             !( it->has_flag( "PURIFIER" ) ) ) {
+            add_msg(_("You can't inject that!"));
+            return true;
+        }
+
+        int mut_str = 0;
+        if ( !it->has_flag( "SERUM" ) ) {
+            mut_str = 1;
+        }
+        else {
+            mut_str = rng(1, 3);
+        }
+        if ( it->has_flag( "PURIFIER" ) ) {
+            mut_str *= -1;
+        }
+        if ( it->has_flag( "MUTAGEN_CEPHALOPOD" ) ||
+            it->has_flag( "PURIFIER" ) ) {
+
+            std::vector<mtype_id> dog_ceph_mut;
+            dog_ceph_mut.push_back(mtype_id("mon_dog"));
+            dog_ceph_mut.push_back(mtype_id("mon_beakhound"));
+            dog_ceph_mut.push_back(mtype_id("mon_beakhound2"));
+            dog_ceph_mut.push_back(mtype_id("mon_beakhound3"));
+            dog_ceph_mut.push_back(mtype_id("mon_beakhound4"));
+
+            int max_mut_lev = dog_ceph_mut.size() - 1;
+
+            for ( size_t i = 0; i < dog_ceph_mut.size(); i++ ) {
+                if (dog_ceph_mut[i] == z->type->id) {
```

Comment:
You could invert that `if` like that:
```
if( dog_ceph_mut[i] != z->type->id ) {
    continue;
}
```

No need to have the entire `for` block in another `if` block.